lib/message_format/parser.rb

```diff
@@ -119,12 +121,74 @@ def parse_text ( parent_type )
       text
     end
 
+    def parse_tag
+      @index += 1
```

Comment:
This is super clever. But it might be simpler in the long run to parse and handle using Nokogiri::HTML5.fragment(string)

Reply:
It might be, but would the nokogiri handle if there is a variable inside of the html like `<TAG>Hello {user}</TAG>`?

Reply:
It could really only handle the parsing/mutating html part. But you could use it to transform:
'<a>{foo}</a>' into '<b>{foo}</b>'

Assuming you're trying to replicate formatjs' interface, you may want to have a look through their tests as I'm sure it has seen a lot of use: https://github.com/formatjs/formatjs/blob/404aaa325f7aee062927ac212150c4b697362fef/packages/intl-messageformat/tests/index.test.ts#L587-L728

Reply:
Maybe some useful insight could be gleaned from their implementation too: https://github.com/formatjs/formatjs/blob/404aaa325f7aee062927ac212150c4b697362fef/packages/icu-messageformat-parser/parser.ts#L327-L348

Which actually kinda reminds me of yours :) so that's encouraging

Reply:
I did look at theirs. Took some inspiration from it and adapted it to how this package was handling parsing already

Reply:
Nice work, @ejohnson6

Reply:
Those tests exposed a couple of small bugs!